hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -499,7 +499,7 @@ private Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> fetchFromSourc
       return new HoodieAvroRecord<>(keyGenerator.getKey(record), payload);
     });
 
-    return Pair.of(schemaProvider, Pair.of(checkpointStr, records));
+    return new ReadResult(schemaProvider, checkpointStr, records, false);
```

Comment:
Can we not omit `avroRDDOptional.get().isEmpty()` check in L484 by pulling up `DataSourceUtils.dropDuplicates` call in this method? Then we can check `isEmpty` at this L502 while instantiating `ReadResult`. It will still be just one `isEmpty` call.

Reply:
let's sync up directly. may be we can resolve it quickly.

Reply:
synced up offline. we end up calling isEmpty twice only when filterDupes is set. for most common use-case, we just call it once.